app/assets/javascripts/discourse-common/addon/lib/deprecated.js

```diff
@@ -70,6 +70,9 @@ export async function withSilencedDeprecations(deprecationIds, callback) {
   try {
     idArray.forEach((id) => disabledDeprecations.add(id));
     result = callback();
+    if (result instanceof Promise) {
+      result = await result;
+    }
```

Comment:
If it's safe to always await:

```suggestion
    result = await callback();
```

Reply:
This was my initial implementation, but unfortunately it doesn't work. It causes the non-async tests to fail because of the change in execution order. (`withSilencedDeprecations` is designed to work with both sync and async stuff)

Reply:
`withSilencedDeprecations` is already an async function, so it always returns a promise. does an extra `await` on non-promise results really change the timing? 👀

Reply:
Yep, if we make that change, one of the `lib/deprecated` tests will start failing.

> withSilencedDeprecations is already an async function, so it always returns a promise

That's true, but consumers don't always `await` that promise.

TBH... this is all very confusing. I was trying to make a single function which could be used in both async and non-async ways. Maybe I should just make a separate `asyncAwaitDeprecation` function for async stuff 👀

Reply:
Alternatively it could be non-async function with promise support. Maybe? 😂 

```js
export function withSilencedDeprecations(deprecationIds, callback) {
  const idArray = [].concat(deprecationIds);
  let result;
  let isAsync;

  idArray.forEach((id) => disabledDeprecations.add(id));

  try {
    result = callback();
    if (result instanceof Promise) {
      isAsync = true;
      result = result.finally(() => {
        idArray.forEach((id) => disabledDeprecations.delete(id));
      });
    }
  } finally {
    if (!isAsync) {
      idArray.forEach((id) => disabledDeprecations.delete(id));
    }
    return result;
  }
}
```